socialdistribution/bettersocial/views.py

```diff
@@ -154,7 +154,23 @@ def get_context_data(self, **kwargs):
             context['posts'] = author.post_set.all()
         else:
             context['author_following_user'] = bool(Following.objects.filter(author = author_uuid, following_uuid = user_uuid))
-            context['user_following_author'] = bool(Following.objects.filter(author = user_uuid, following_uuid = author_uuid))
+            if Author.objects.filter(uuid = user_uuid).exists():
+                context['user_following_author'] = bool(Following.objects.filter(author = user_uuid, following_uuid = author_uuid))
+            else:
+                # TODO: make sure this works
+                node = Node.objects.filter(host__contains = self.request.GET['host']).get()
+                url = yarl.URL(node.host + '/author/' + author_uuid + '/followers/' + user_uuid).human_repr()
+                response = requests.get(
+                    url,
+                    headers = { 'Accept': 'application/json' },
+                    auth = HTTPBasicAuth(node.node_username, node.node_password),
+                )
+
+                if response.ok:
+                    print(response.content)
+                    context['user_following_author'] = bool(Following.objects.filter(author = user_uuid, following_uuid = author_uuid))
+                else:
+                    context['user_following_author'] = False
 
             # TODO: Might only need to have Public posts to be queried or publick and friends posts?
             context['posts'] = Post.objects.filter(
```

Comment:
You'll have to change the following lines in order to grab the posts of the remote user, as it would throw an error because a remote UUID will not be present in our authors table.

That being said that's information likely useful for your other PR — not this one

Reply:
I'll move this to the local user section for now and add a todo for later